database/migrations/2025_12_29_000000_update_useraccounts_table.php

```diff
@@ -0,0 +1,277 @@
+                $table->renameIndex('useraccounts_lastlogin_deleted_index', 'users_last_activity_at_deleted_at_index');
+                $table->renameIndex('useraccounts_visible_role_id_index', 'users_visible_role_id_index');
+            });
+        }
+
+        // Reorder columns into logical groups.
+        if (DB::connection()->getDriverName() !== 'sqlite') {
+            DB::statement(<<<SQL
+                ALTER TABLE users
+                    -- Identity
+                    MODIFY COLUMN `ulid` char(26) DEFAULT NULL AFTER `id`,
+                    MODIFY COLUMN `username` varchar(32) NOT NULL AFTER `ulid`,
+                    MODIFY COLUMN `display_name` varchar(255) DEFAULT NULL AFTER `username`,
+
+                    -- Authentication
+                    MODIFY COLUMN `password` varchar(255) DEFAULT NULL AFTER `display_name`,
+                    MODIFY COLUMN `two_factor_secret` text DEFAULT NULL AFTER `password`,
+                    MODIFY COLUMN `two_factor_recovery_codes` text DEFAULT NULL AFTER `two_factor_secret`,
+                    MODIFY COLUMN `two_factor_confirmed_at` timestamp NULL DEFAULT NULL AFTER `two_factor_recovery_codes`,
+                    MODIFY COLUMN `legacy_salted_password` varchar(32) NOT NULL AFTER `two_factor_confirmed_at`,
+                    MODIFY COLUMN `remember_token` varchar(100) DEFAULT NULL AFTER `legacy_salted_password`,
+
+                    -- Email
+                    MODIFY COLUMN `email` varchar(64) NOT NULL AFTER `remember_token`,
+                    MODIFY COLUMN `email_verified_at` timestamp NULL DEFAULT NULL AFTER `email`,
+                    MODIFY COLUMN `email_backup` varchar(255) DEFAULT NULL AFTER `email_verified_at`,
+
+                    -- Authorization
+                    MODIFY COLUMN `visible_role_id` bigint(20) unsigned DEFAULT NULL AFTER `email_backup`,
+                    MODIFY COLUMN `Permissions` tinyint(4) NOT NULL COMMENT '-2=spam, -1=banned, 0=unconfirmed, 1=confirmed, 2=jr-developer, 3=developer, 4=moderator' AFTER `visible_role_id`,
+
+                    -- Player stats
+                    MODIFY COLUMN `achievements_unlocked` int(10) unsigned DEFAULT NULL AFTER `Permissions`,
+                    MODIFY COLUMN `achievements_unlocked_hardcore` int(10) unsigned DEFAULT NULL AFTER `achievements_unlocked`,
+                    MODIFY COLUMN `completion_percentage_average` decimal(10,9) DEFAULT NULL AFTER `achievements_unlocked_hardcore`,
+                    MODIFY COLUMN `completion_percentage_average_hardcore` decimal(10,9) DEFAULT NULL AFTER `completion_percentage_average`,
+                    MODIFY COLUMN `points` int(11) NOT NULL AFTER `completion_percentage_average_hardcore`,
+                    MODIFY COLUMN `points_softcore` int(11) DEFAULT 0 AFTER `points`,
+                    MODIFY COLUMN `points_weighted` int(10) unsigned DEFAULT NULL AFTER `points_softcore`,
```

Comment:
Recommend ordering these as `points_softcore`, `points`, and `points_weighted` so the weighted points are next to the hardcore points. It's also consistent with softcore/hardcore ordering on other fields like `achievements_unlocked`.